grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -198,7 +198,7 @@ class ProductController {
 
         ProductType defaultProductType = ProductType.defaultProductType.list()?.first();
         // Throw an error for product type with empty code and product identifier that is not a default product type
-        if (productInstance.productType != defaultProductType && !productInstance.productType?.code && !productInstance.productType?.productIdentifierFormat) {
+        if (productInstance.productType?.id != defaultProductType?.id && !productInstance.productType?.code && !productInstance.productType?.productIdentifierFormat) {
```

Comment:
addressed the primitive value comparison instead of the whole `ProductType` object 
based on the discussion from https://github.com/openboxes/openboxes/pull/3541#discussion_r994072420